### PR TITLE
MF-16 notify employer new reservation

### PR DESCRIPTION
**Pull request by scottywakefield**

### CODE REVIEW

File: src/SFA.DAS.Reservations.Application.UnitTests/Reservations/Services/WhenNotifyingEmployerOfNewReservation.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            NotificationsService service)
+        {
+            Dictionary<string, string> expectedTokens = null;
+
+            service.SendNewReservationMessage(createdMessage);
+            
+            mockNotificationClient.Verify(api => api.SendEmail(It.Is<Email>(email => 
+                email.RecipientsAddress == createdMessage.RecipientsAddress &&
+                //email.ReplyToAddress == "todo" && //todo: already set in template?
+                //email.Subject == "Apprenticeship service: funding reservation made on your behalf" && //todo: needed?
```
Comment (robinoconnor) [inline src/SFA.DAS.Reservations.Application.UnitTests/Reservations/Services/WhenNotifyingEmployerOfNewReservation.cs]: Is there any stories to pick up these todos or is it better to get them answered and sorted for this story?

File: src/SFA.DAS.Reservations.Application/Reservations/Handlers/ReservationCreatedHandler.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+            var users = await _accountsService.GetAccountUsers(createdEvent.AccountId);
+
+            _logger.LogDebug($"Account [{createdEvent.AccountId}] has [{users.Count()}] users in total.");
+
+            var filteredUsers = users.Where(user => 
+                user.CanReceiveNotifications && 
+                _permittedRoles.Contains(user.Role));
+
+            _logger.LogDebug($"Account [{createdEvent.AccountId}] has [{users.Count()}] users with correct role and subscription.");
```
Comment (robinoconnor) [inline src/SFA.DAS.Reservations.Application/Reservations/Handlers/ReservationCreatedHandler.cs]: Shouldn't the count be for Filtered users?

File: src/SFA.DAS.Reservations.Infrastructure/QueueNames.cs [hunk 3]
```diff
@@ -10,5 +10,6 @@ public static class QueueNames
         public const string SignedAgreement = "SFA.DAS.Reservations.Jobs.SignedAgreement";
         public const string DraftApprenticeshipDeleted = "SFA.DAS.Reservations.Jobs.ApprenticeshipDeleted";
         public const string LevyAddedToAccount = "SFA.DAS.Reservations.Jobs.LevyAddedToAccount";
+        public const string ReservationCreated = "SFA.DAS.Reservations.Jobs.ReservationCreated";//todo: is this correct??
```
Comment (robinoconnor) [inline src/SFA.DAS.Reservations.Infrastructure/QueueNames.cs]: Do this need to be worked out before the branch is merged?

File: src/SFA.DAS.Reservations.Functions.Reservations/Startup.cs [hunk 4]
```diff
@@ -92,12 +100,25 @@ public IServiceProvider Build()
...
+
             services.AddTransient<IReservationService,ReservationService>();
+            services.AddTransient<IProviderService, ProviderService>();
+            services.AddTransient<IAccountsService, AccountsService>();
+            services.AddTransient<INotificationsService, NotificationsService>();
+
+            services.AddTransient<INotificationTokenBuilder, NotificationTokenBuilder>();
             services.AddTransient<IReservationRepository,ReservationRepository>();
 
+            //todo: get encoding config.
```
Comment (robinoconnor) [inline src/SFA.DAS.Reservations.Functions.Reservations/Startup.cs]: Do these todos need to be resolved before its merged?

File: src/SFA.DAS.Reservations.Application.UnitTests/SFA.DAS.Reservations.Application.UnitTests.csproj [hunk 5]
```diff
@@ -1,20 +1,26 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
```
Comment (ChughAkshay) [inline src/SFA.DAS.Reservations.Application.UnitTests/SFA.DAS.Reservations.Application.UnitTests.csproj]: framework upgraded (I assume this is was an intentional change?)

File: src/SFA.DAS.Reservations.Data/SFA.DAS.Reservations.Data.csproj [hunk 6]
```diff
@@ -4,8 +4,8 @@
     <TargetFramework>netstandard2.0</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="2.1.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.1.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="2.2.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.2.0" />
```
Comment (ChughAkshay) [inline src/SFA.DAS.Reservations.Data/SFA.DAS.Reservations.Data.csproj]: framework upgrade - just highlighting in case unintentional

Comment (scottywakefield) [reply src/SFA.DAS.Reservations.Data/SFA.DAS.Reservations.Data.csproj]: thanks man, yeah, one of the packages is for api we use is dependant on 2.2.

**Pull request merged**